language/move-vm/runtime/src/runtime.rs

```diff
@@ -339,6 +349,102 @@ impl VMRuntime {
         Ok(())
     }
 
+    pub(crate) fn execute_function_for_effects(
```

Comment:
This could probably share almost all its code with `execution_function_impl`, but wrote it out directly for readability first.